src/third_party/dartino/gc_metadata.cc

```diff
@@ -22,7 +22,7 @@ void GcMetadata::set_up() { singleton_.set_up_singleton(); }
 void GcMetadata::set_up_singleton() {
   OS::HeapMemoryRange range = OS::get_heap_memory_range();
 
-  lowest_address_ = reinterpret_cast<uword>(range.address);
+  lowest_address_ = Utils::round_down(reinterpret_cast<uword>(range.address), TOIT_PAGE_SIZE);
```

Comment:
The lowest bound of the allocated memory must be aligned for calculations that find metadata from address to work right.